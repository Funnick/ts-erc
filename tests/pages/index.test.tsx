import "@testing-library/jest-dom";
import { render, screen, act } from "@testing-library/react";
import renderer from 'react-test-renderer';
import Home from "@/pages";
import fetch from 'jest-fetch-mock';

describe("Home page test", () => {
    it("renders a header", async () => {
      fetch.mockResponseOnce(JSON.stringify({ rates: {"CAD": 1.42, "EUR": 1, "USD": 1.11}, base: "EUR" }))

      await act(async () => render(<Home />));

      const title = screen.getByText("Currency Converter")
      const text = screen.getByText("Enter Amount")
      const input = screen.getByRole('spinbutton')
      const button = screen.getByRole('button')
      
      expect(title).toBeInTheDocument()
      expect(text).toBeInTheDocument()
      expect(input).toBeInTheDocument()
      expect(button).toBeInTheDocument()
    });

    it('Snapshot', async () => {
      fetch.mockResponseOnce(JSON.stringify({ rates: {"CAD": 1.42, "EUR": 1, "USD": 1.11}, base: "EUR" }))

      const tree = await act(async () => renderer.create(<Home />).toJSON());

      expect(tree).toMatchSnapshot()
  })
});