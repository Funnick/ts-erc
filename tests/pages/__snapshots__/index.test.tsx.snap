// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page test Snapshot 1`] = `null`;
